parts/k8s/windowskubeletfunc.ps1

```diff
@@ -605,7 +660,161 @@ catch
 }
 
 "@
-    } # end else using WinCNI.
+    } # end elseif using WinCNI and Docker.
+    elseif (($NetworkPlugin -eq "kubenet" ) -and ($UseContainerD -eq $true))
+    {
+        # TODO: something is wrong with the CNI configuration
+        # Warning  FailedCreatePodSandBox  2m16s (x222 over 50m)  kubelet, 4068k8s011  (combined from similar events): Failed to create pod sandbox: 
+        # rpc error: code = Unknown desc = failed to setup network for sandbox "922b1a200078edb15c7a5732612cbe19e5dadf7cf8a4622d72b376874522435d": error creating endpoint hcnCreateEndpoint failed in Win32: Invalid JSON document string. (0x803b001b) {"Success":false,"Error":"Invalid JSON document string. {{Policies.Type,UnknownEnumValue}}","ErrorCode":2151350299} : endpoint config &{ 922b1a200078edb15c7a5732612cbe19e5dadf7cf8a4622d72b376874522435d_l2bridge f94e9649-d4df-486e-bcd4-2721938d89f3  [{OutBoundNAT []} {ROUTE []}] [] { [default.svc.cluster.local] [10.0.0.10] []} [{10.240.0.1 0.0.0.0/0 0}]  0 {2 0}}
+
+
+        # using SDNBridge & Containerd
+        $KubeNetwork = "l2bridge"
+        $kubeStartStr += @"
+
+function
+Get-DefaultGateway(`$CIDR)
+{
+    return `$CIDR.substring(0,`$CIDR.lastIndexOf(".")) + ".1"
+}
+
+function
+Get-PodCIDR()
+{
+    `$podCIDR = c:\k\kubectl.exe --kubeconfig=c:\k\config get nodes/`$(`$env:computername.ToLower()) -o custom-columns=podCidr:.spec.podCIDR --no-headers
+    return `$podCIDR
+}
+
+function
+Test-PodCIDR(`$podCIDR)
+{
+    return `$podCIDR.length -gt 0
+}
+
+function
+Update-CNIConfig(`$podCIDR, `$masterSubnetGW)
+{
+    `$jsonSampleConfig =
+"{
+    ""cniVersion"": ""0.2.0"",
+    ""name"": ""<NetworkMode>"",
+    ""type"": ""sdnbridge.exe"",
+    ""master"": ""Ethernet"",
+    ""capabilities"": { ""portMappings"": true },
+    ""ipam"": {
+        ""environment"": ""azure"",
+        ""subnet"":""<PODCIDR>"",
+        ""routes"": [{
+        ""GW"":""<PODGW>""
+        }]
+    },
+    ""dns"" : {
+    ""Nameservers"" : [ ""<NameServers>"" ],
+    ""Search"" : [ ""<Cluster DNS Suffix or Search Path>"" ]
+    },
+    ""AdditionalArgs"" : [
+    {
+        ""Name"" : ""EndpointPolicy"", ""Value"" : { ""Type"" : ""OutBoundNAT"", ""Settings"" : { ""Exceptions"": [ ""<ClusterCIDR>"", ""<MgmtSubnet>"" ] }}
+    },
+    {
+        ""Name"" : ""EndpointPolicy"", ""Value"" : { ""Type"" : ""SDNRoute"", ""Settings"" : { ""DestinationPrefix"": ""<ServiceCIDR>"", ""NeedEncap"" : true }}
+    }
+    ]
+}"
+
+    `$configJson = ConvertFrom-Json `$jsonSampleConfig
+    `$configJson.name = `$global:NetworkMode.ToLower()
+    `$configJson.ipam.subnet=`$podCIDR
+    `$configJson.ipam.routes[0].GW = `$masterSubnetGW
+    `$configJson.dns.Nameservers[0] = `$global:KubeDnsServiceIp
+    `$configJson.dns.Search[0] = `$global:KubeDnsSearchPath
+
+
+    `$configJson.AdditionalArgs[0].Value.Settings.Exceptions[0] = `$global:KubeClusterCIDR

```

Comment:
@ksubrmnn - this is what I mentioned - needed to change this to match the updated schema in `$jsonSampleConfig`

Reply:
Oh man that’s frustrating. I changed it in the set up you gave me but dropped it in the new branch. Couldn’t test with no subscription :/ 

Didn’t mean to give you the run around!